lib/methods/veres-one/client.js

```diff
@@ -0,0 +1,318 @@
+/*!
+ * Copyright (c) 2018 Digital Bazaar, Inc. All rights reserved.
+ */
+'use strict';
+
+const url = require('url');
+const r2 = require('r2');
+// const https = require('https');
+
+const DEFAULT_LOCATION = 'ledger';
+const DEFAULT_MODE = 'test';
+
+class VeresOneClient {
+  constructor(options = {}) {
+    this.config = options.config || require('./veres-one.config');
+  }
+
+  /**
+   * Fetches a DID Document for a given DID.
+   *
+   * @param did {string}
+   * @param [mode] {string} One of 'dev'/'test'/'live'
+   * @param [hostname] {string} Optional hostname of ledger (falls back
+   *   to the default hostname for a particular mode)
+   *
+   * @returns {Promise<object>} Resolves to DID Document Fetch Result
+   */
+  async get({did, mode, hostname}) {
+    const ledgerAgent = await this.getAgent({mode, hostname});
+    const queryServiceUrl = url.parse(ledgerAgent.service.ledgerQueryService);
+    queryServiceUrl.search = `id=${did}`;
+    const didUrl = url.format(queryServiceUrl);
+
+    // const https = require('https');
+    // let agent;
+    // if(options.mode === 'dev') {
+    //   agent = new https.Agent({rejectUnauthorized: false});
+    // }
+    // _verbose(options, 'Retrieving remote DID Document...', {hostname});
+    // _debug(options, 'remote request', {url: didUrl});
+
+    const response = await r2({
+      url: didUrl,
+      method: 'POST',
+      // agent,
+      headers: {
+        'accept': 'application/ld+json, application/json'
+      }
+    }).response;
+
+    if(response.status !== 200) {
```

Comment:
note, `axios` throws for any non-success status code, so one gets this sort of functionality for free just by surrounding api calls with try/catch.